server/store/alert.go

```diff
@@ -0,0 +1,98 @@
+package store
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	splunkAlertKey = "splunkalert"
+	splunkAlertMap = "splunkalertmap"
+)
+
+// AlertStore API for alert KVStore.
+type AlertStore interface {
+	GetAlertChannelID(alert string) (string, error)
+	GetChannelAlertIDs(channelID string) ([]string, error)
+	CreateAlert(channelID string, alertsID string) error
+	DeleteChannelAlert(channelID string, alertsID string) error
+}
+
+func keyWithChannelID(key, id string) string {
+	return fmt.Sprintf("%s_%s", key, id)
+}
+
+func (s *pluginStore) GetAlertChannelID(alertID string) (string, error) {
+	var alertsMap map[string]string
+	err := loadJSON(s.alertStore, splunkAlertMap, &alertsMap)
+	if err != nil {
+		return "", err
+	}
+	if channelID, ok := alertsMap[alertID]; ok {
+		return channelID, nil
+	}
+	return "", nil
+}
+
+func (s *pluginStore) GetChannelAlertIDs(channelID string) ([]string, error) {
+	var alerts []string
+	err := loadJSON(s.alertStore, keyWithChannelID(splunkAlertKey, channelID), &alerts)
+	return alerts, err
+}
+
+func (s *pluginStore) CreateAlert(channelID string, alertID string) error {
+	channelAlerts, err := s.GetChannelAlertIDs(channelID)
+	if err != nil {
+		return err
+	}
+	var alertsMap = make(map[string]string)
+	err = loadJSON(s.alertStore, splunkAlertMap, &alertsMap)
+	if err != nil {
+		return err
+	}
+	alertsMap[alertID] = channelID
+	err = setJSON(s.alertStore, splunkAlertMap, alertsMap)
+	if err != nil {
+		return err
+	}
+	channelAlerts = append(channelAlerts, alertID)
+	err = setJSON(s.alertStore, keyWithChannelID(splunkAlertKey, channelID), channelAlerts)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *pluginStore) DeleteChannelAlert(channelID string, alertID string) error {
+	subscriptions, err := s.GetChannelAlertIDs(channelID)
+	if err != nil {
+		return err
+	}
+	var alertsMap = make(map[string]string)
+	err = loadJSON(s.alertStore, splunkAlertMap, &alertsMap)
+	if err != nil {
+		return err
+	}
+	subIndex := findInSlice(subscriptions, alertID)
```

Comment:
Is the code really more readable with this utils method?

Reply:
this function was already exist so thought of using it